integration/proxy_tunnel_strategy_test.go

```diff
@@ -0,0 +1,422 @@
+
+		creds, err := GenerateUserCreds(UserCredsRequest{
+			Process:  p.auth.Process,
+			Username: p.username,
+		})
+		require.NoError(t, err)
+
+		client, err := proxy.NewClientWithCreds(
+			ClientConfig{
+				Cluster: p.cluster,
+				Host:    nodeuuid,
+			},
+			*creds,
+		)
+		require.NoError(t, err)
+
+		output := &bytes.Buffer{}
+		client.Stdout = output
+
+		cmd := []string{"echo", "hello world"}
+		err = client.SSH(context.Background(), cmd, false)
+		require.NoError(t, err)
+		require.Equal(t, "hello world\n", output.String())
+	}
+}
+
+func (p *proxyTunnelStrategy) dialDatabase(t *testing.T) {
+	for i, proxy := range p.proxies {
+		connClient, err := postgres.MakeTestClient(context.Background(), common.TestClientConfig{
+			AuthClient: p.dbAuthClient,
+			AuthServer: p.auth.Process.GetAuthServer(),
+			Address:    proxy.GetWebAddr(),
+			Cluster:    p.cluster,
+			Username:   p.username,
+			RouteToDatabase: tlsca.RouteToDatabase{
+				ServiceName: p.cluster + "-postgres",
+				Protocol:    defaults.ProtocolPostgres,
+				Username:    "postgres",
+				Database:    "test",
+			},
+		})
+		require.NoError(t, err)
+
+		result, err := connClient.Exec(context.Background(), "select 1").ReadAll()
+		require.NoError(t, err)
+		require.Equal(t, []*pgconn.Result{postgres.TestQueryResponse}, result)
+		require.Equal(t, uint32(i+1), p.postgresDB.QueryCount())
+
+		err = connClient.Close(context.Background())
+		require.NoError(t, err)
+	}
+}
+
+// makeLoadBalancer bootsraps a new load balancer for proxy instances.
+func (p *proxyTunnelStrategy) makeLoadBalancer(t *testing.T) {
+	lbAddr := utils.MustParseAddr(net.JoinHostPort(Loopback, strconv.Itoa(ports.PopInt())))
+	lb, err := utils.NewLoadBalancer(context.Background(), *lbAddr)
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		require.NoError(t, lb.Close())
+	})
+
+	require.NoError(t, lb.Listen())
+	go lb.Serve()
+
+	if p.lb == nil {
+		p.lb = lb
+	} else {
+		t.Error("load balancer already initialized")
+	}
+}
+
+// makeAuth bootsraps a new teleport auth instance.
+func (p *proxyTunnelStrategy) makeAuth(t *testing.T, strategy *types.TunnelStrategyV1) {
+	privateKey, publicKey, err := testauthority.New().GenerateKeyPair("")
+	require.NoError(t, err)
+
+	auth := NewInstance(InstanceConfig{
+		ClusterName: p.cluster,
+		HostID:      uuid.New().String(),
+		NodeName:    Loopback,
+		Priv:        privateKey,
+		Pub:         publicKey,
+		log:         utils.NewLoggerForTests(),
+	})
+
+	auth.AddUser(p.username, []string{p.username})
+
+	conf := service.MakeDefaultConfig()
+	conf.DataDir = t.TempDir()
+	conf.Auth.Enabled = true
+	conf.Auth.NetworkingConfig.SetTunnelStrategy(strategy)
+	conf.Proxy.Enabled = false
+	conf.SSH.Enabled = false
+
+	require.NoError(t, auth.CreateEx(t, nil, conf))
+
+	t.Cleanup(func() {
+		require.NoError(t, auth.StopAll())
+	})
+	require.NoError(t, auth.Start())
+
+	if p.auth == nil {
+		p.auth = auth
+	} else {
+		t.Error("auth already initialized")
+	}
```

Comment:
Same: check for `p.auth != nil` at the beginning of the function and change from `t.Error` to `require.Fail`.